application_test.go

```diff
@@ -42,6 +42,32 @@ func TestApplicationMemory(t *testing.T) {
 	assert.Equal(t, 50.0, *app.Mem)
 }
 
+func TestNewApplicationString(t *testing.T) {
```

Comment:
Could you please try to merge this test with `TestApplicationString` in a table-driven fashion like [we also do it here](https://github.com/gambol99/go-marathon/blob/3a8544a7c28ec83645fc96feff75bc3174e426aa/unreachable_strategy_test.go#L96-L129)?

I think the test struct should come with a `func(app *Application)` that does the version-specific modifications, and another string specifying the golden test data to compare against.

Please also try to do so for the other newly created tests that are similar enough to existing tests.